### PR TITLE
feat: add C implementation for `math/base/special/cosm1`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/cosm1`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/cosm1).

```c
double stdlib_base_cosm1( const double x );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/cosm1/test/test.native.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+// TESTS //
+
+tape( 'main export is a function', opts, function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof cos, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function computes the cosine minues one more accurately inside the interval [-π/4,π/4]', opts, function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/cosm1/test/test.native.js]: Would you mind checking for and potentially fixing this typo also in the main `test.js` file, @gunjjoshi?

```suggestion
tape( 'the function computes the cosine minus one more accurately inside the interval [-π/4,π/4]', opts, function test( t ) {
```

**Review by Planeshifter — APPROVED** [review]
Noticed only one typo in a test description; all else looks great!

File: lib/node_modules/@stdlib/math/base/special/cosm1/test/test.native.js (hunk 1 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/cosm1/test/test.native.js]: @Planeshifter Done!

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/cosm1/test/test.native.js]: Thanks!

**Pull request merged**